### DISCUSSION TITLE
Prettifying JSON Output

### DISCUSSION THREAD

**Discussion by andrei27m** [trpc/trpc#3745] [Q&A]
During development it would be helpful to have trpc return prettified JSON output, especially in cases where the JSON output is some large block of text that's difficult to parse visually.

I thought transformers would be the way to go, so I added this:
```typescript
const t = initTRPC.meta<OpenApiMeta>().context<Context>().create({
    transformer: {
        serialize: (object: any): string => {
            return JSON.stringify(object, undefined, 2)
        },
        deserialize(object: any): any {
            return JSON.parse(object)
        }
    }
})
```
But neither `serialize` or `deserialize` seem to ever be called. What am I doing wrong?

**Comment by arosemena** [comment]
Have you tried with `input` and `output` keys inside `transformer` as the example shows? https://trpc.io/docs/data-transformers

Maybe something like this?

```ts
const t = initTRPC.meta<OpenApiMeta>().context<Context>().create({
  transformer: {
    output: {
      serialize: (object: any): string => {
        return JSON.stringify(object, undefined, 2)
      },
      deserialize(object: any): any {
        return JSON.parse(object)
      },
    },
  },
})
```
